### PR TITLE
[Shepherd] LCM API implemented with dictionaries and JSON

### PR DESCRIPTION
**Pull request by charleshong3**
Added address for LCM objects to use, so LCM works over ethernet.
Uses python dictionaries as input and output for functions.
Uses JSON formatted string to send and receive over LCM.

### CODE REVIEW

File: shepherd/LCM.py [hunk 1]
```diff
@@ -1,26 +1,23 @@
 import threading
-import lcm # pylint: disable=import-error
+import json
+import lcm
+
+LCM_address = 'udpm://239.255.76.68:7667?ttl=2'
```
Comment (JHGZhou) [inline shepherd/LCM.py]: when was this added and is it necessary?

File: shepherd/LCM.py [hunk 2]
```diff
@@ -32,13 +29,10 @@ def run():
     rec_thread.run = run
     rec_thread.start()
 
-def lcm_send(target_channel, header, *args):
+def lcm_send(target_channel, header, dic):
```
Comment (JHGZhou) [inline shepherd/LCM.py]: might be a bit confusing, but api should support not having the `dic` argument filled in. in that case it should just send an empty dictionary

File: shepherd/LCM.py [hunk 3]
```diff
@@ -1,26 +1,23 @@
 import threading
-import lcm # pylint: disable=import-error
+import json
+import lcm
```
Comment (JHGZhou) [inline shepherd/LCM.py]: you need to add the pylint disable from before

**Review by JHGZhou — CHANGES_REQUESTED** [review]
made 3 comments, should be quick resolutions

File: shepherd/LCM.py (hunk 1 above)
Comment (charleshong3) [reply shepherd/LCM.py]: You need an address to send messages over ethernet/through a switch. If these methods aren't used for that then it isn't necessary.

**Pull request closed without merging**